src/main/scala/chisel3/util/Mux.scala

```diff
@@ -17,6 +18,21 @@ import chisel3._
   *  io.selector(2) -> 8.U,
   *  io.selector(4) -> 11.U,
   * ))
+  *
+  * // or using ChiselEnum1H
+  *
+  * object selector extends chisel3.experimental.ChiselEnum1H {
+  *   val s0, s1, s2, s3 = Value
+  * }
+  * val selectorVal = Wire(UInt(selector.getWidth.W))
+  * selectorVal := selector.s1.asUInt
+  * val hotValue = chisel3.util.Mux1H(selectorVal.asUInt,
+      Seq(
+  *     2.U,
+  *     4.U,
+  *     8.U,
+  *     11.U,
```

Comment:
I think we need add a sugar for Mux1H to consume a Enum type and return a Enum type.
These magic num is strange, while seems not correct: `11.U -> 16.U`

Reply:
This can be done in future PR(just comment for recording)

Reply:
I think we need add a sugar for Mux1H to consume a Enum type and return a Enum type.
These magic num is strange, while seems not correct: 11.U -> 16.U